crates/sui-core/src/execution_cache.rs

```diff
@@ -755,9 +776,13 @@ macro_rules! implement_storage_traits {
                 // transaction replay due to possible reordering of transactions during replay.
                 if recv_object.owner != Owner::AddressOwner((*owner).into())
                     || self.have_received_object_at_version(
-                        receiving_object_id,
-                        receive_object_at_version,
+                        // TODO-DNS Is it safe to assume receiving objects are always fastpath objects?
```

Comment:
i think we will have to support receiving objects that are address-owned by an object id, whether or not they are fast-path or consensus v2. It would be very odd to say "if you send object A to object B as a fastpath object, it can be received, but if you send it as a consensus v2 object you've burned it".

(i don't think we need to launch with this support since we aren't yet going to advertise consensus v2 as the preferred mechanism, but we will probably want it eventually)